### PR TITLE
include url and saml connector name in entity descriptor url errors

### PR DESCRIPTION
**Pull request by stevenGravy**
Errors can occur if a SAML entity descriptor url is no longer available.  Without knowing the connector name it's difficult to know which SAML connector is having the problem just from the entity descriptor url.

```bash
tctl get connectors
```

If a entity descriptor url fails you get the error:
```
ERROR: rpc error: code = Unknown desc = Get "https://gravitational-preview.oktapreview.com/app/exk7073it1Ywz0t8L1d7/sso/saml/metadata": dial tcp 4.4.4.4:443: i/o timeout
```

Updated to this so you get the url and the saml connector name `okta`. 

```
ERROR: rpc error: code = Unknown desc = error getting entitydescriptor from "https://gravitational-preview.oktapreview.com/app/exk7073it1Ywz0t8L1d7/sso/saml/metadata" for SAML connector:okta
        Get "https://gravitational-preview.oktapreview.com/app/exk7073it1Ywz0t8L1d7/sso/saml/metadata": dial tcp 4.4.4.4:443: i/o timeout
```

### CODE REVIEW

**Review by jentfoo — APPROVED** [review]

**Review by avatus — APPROVED** [review]

File: lib/services/saml.go [hunk 1]
```diff
@@ -49,18 +49,18 @@ func ValidateSAMLConnector(sc types.SAMLConnector, rg RoleGetter) error {
 	if sc.GetEntityDescriptorURL() != "" {
 		resp, err := http.Get(sc.GetEntityDescriptorURL())
 		if err != nil {
-			return trace.Wrap(err)
+			return trace.WrapWithMessage(err, "error getting entitydescriptor from %q for SAML connector:%v", sc.GetEntityDescriptorURL(), sc.GetName())
```
Comment (zmb3) [inline lib/services/saml.go:52]: entity descriptor is two words right?


Also, do we need to prefix the message with _error_? Won't it be pretty obvious that it's an error?

Comment (stevenGravy) [reply lib/services/saml.go]: Please see update @zmb3

File: lib/services/saml.go [hunk 2]
```diff
@@ -49,18 +49,18 @@ func ValidateSAMLConnector(sc types.SAMLConnector, rg RoleGetter) error {
 	if sc.GetEntityDescriptorURL() != "" {
 		resp, err := http.Get(sc.GetEntityDescriptorURL())
 		if err != nil {
-			return trace.Wrap(err)
+			return trace.WrapWithMessage(err, "unable to fetch entity descriptor from %q for SAML connector:%v", sc.GetEntityDescriptorURL(), sc.GetName())
```
Comment (zmb3) [inline lib/services/saml.go:52]:
```suggestion
			return trace.WrapWithMessage(err, "unable to fetch entity descriptor from %v for SAML connector %v", sc.GetEntityDescriptorURL(), sc.GetName())
```

I don't think we need to quote the URL, and let's add a space after the word connector.

(Same for quoting the URL below too)

Comment (stevenGravy) [reply lib/services/saml.go]: ok updated

Comment (stevenGravy) [reply lib/services/saml.go]: Thanks, all url output updated to `%v`

**Review by zmb3 — APPROVED** [review]

**Pull request merged**